### PR TITLE
Fortran do conc

### PR DESCRIPTION
**Pull request by hagabb**
# Adding a New Sample(s)

## Description

This is the Fortran edge detection sample that was approved by the oneAPI Code Samples WG earlier this year. It implements a simple edge detection algorithm using ordinary for-loops and do concurrent loops, then offloads the computing to a GPU using OpenMP.

This sample is more thoroughly described in a recent article on IDZ: https://www.intel.com/content/www/us/en/developer/articles/technical/using-fortran-do-current-for-accelerator-offload.html.

## Checklist
Administrative
- [ ] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): Paul Petersen
- [ ] If you have any new dependencies/binaries, inform the oneAPI Code Samples Project Manager

Code Development
- [ ] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [ ] Adhere to readme template 
- [ ] Enforce format via clang-format config file
- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [ ] Compile using the following compiler flags and fix any warnings, the flags are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Petersen. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

**Review by jimmytwei — APPROVED** [review]

File: DirectProgramming/Fortran/EdgeDetection/simple-binary-images/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ 1. img_seg_omp_target.F90 implements edge detection on binary images using ordinary for-loops and OpenMP target directives
+ 2. img_seg_do_concurrent.F90 implements edge detection on binary images using only a DO CONCURRENT loop
+
+The implementations are functionally equivalent. In both cases, the OpenMP runtime library is used to parallelize the
+edge detection loops, regardless of whether they are run on the CPU or offloaded to a GPU.
+
+## Key Implementation Details
+[Using Fortran DO CONCURRENT for Accelerator Offload](https://www.intel.com/content/www/us/en/developer/articles/technical/using-fortran-do-current-for-accelerator-offload.html) provides more detailed descriptions of each example code, and discusses the relative merits of each approach.
+
+## License
```
Comment (jimmytwei) [inline DirectProgramming/Fortran/EdgeDetection/simple-binary-images/README.md:30]: @hagabb Can you put the ``License`` section at the end?

File: DirectProgramming/Fortran/EdgeDetection/simple-binary-images/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+> environment by sourcing  the `setvars` script located in
+> the root of your oneAPI installation.
+>
+> Linux Sudo: . /opt/intel/oneapi/setvars.sh
+>
+> Linux User: . ~/intel/oneapi/setvars.sh
+>
+>For more information on environment variables, see Use the setvars Script for [Linux or macOS](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html).
+
+### Running Samples on the DevCloud
```
Comment (jimmytwei) [inline DirectProgramming/Fortran/EdgeDetection/simple-binary-images/README.md:64]: @hagabb Can we remove this section (the one that references DevCloud)? We'll have the correct verbiage for IDC at a later date.

Comment (hagabb) [reply DirectProgramming/Fortran/EdgeDetection/simple-binary-images/README.md]: I removed the section about the IDC.

File: DirectProgramming/Fortran/EdgeDetection/simple-binary-images/README.md (hunk 1 above)
Comment (hagabb) [reply DirectProgramming/Fortran/EdgeDetection/simple-binary-images/README.md]: I moved the License section to the end of the README.

**Review by jimmytwei — APPROVED** [review]

**Pull request merged**